src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -824,7 +824,7 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 			this._onConnectRequestSent.fire();
 
 			// TODO make this generic enough to handle non-SQL languages too
-			this.doChangeLanguageFlavor(uri, 'sql', connection.providerName);
+			this.doChangeLanguageFlavor(uri, this._resourceProviderService._providers[connection.providerName].providerLanguageId, connection.providerName);
```

Comment:
There are some extensions for PostgreSQL and MySQL Db which I believe use the MSSQL as the provider, however the connection logic is different. We do not want to break those extensions, as they will not be updated with a languageId. It might make sense to write a function here like you had. Where we check if languageId is defined and use it, if not, we hard code a mapping where for 'MSSQL' we return 'sql'.

Reply:
Got it. I have added a function to handle it with few explanatory comments.